### PR TITLE
chore: fix some compiler warnings (MINOR)

### PR DESCRIPTION
**Pull request by mjsax**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitorTest.java [hunk 1]
```diff
@@ -470,7 +470,7 @@ public void shouldGenerateCorrectCodeForCaseStatementWith13Conditions() {
                                                        "six", "seven", "eight", "nine", "ten",
                                                        "eleven", "twelve");
 
-    final ImmutableList arg = numbers
+    final ImmutableList<WhenClause> arg = numbers
```
Comment (mjsax) [inline ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitorTest.java:473]: Fixing rawtype

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/ListVariablesExecutorTest.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public void setup() {
   }
 
   private Optional<KsqlEntity> executeListVariables(final String sql) {
-    final ConfiguredStatement configuredStatement = mock(ConfiguredStatement.class);
+    final ConfiguredStatement<?> configuredStatement = mock(ConfiguredStatement.class);
```
Comment (mjsax) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/ListVariablesExecutorTest.java:48]: As above

File: ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/CastEvaluatorTest.java [hunk 3]
```diff
@@ -729,6 +729,7 @@ static Object instanceFor(final SqlType type, final SqlType to) {
           if (to.baseType() == TIMESTAMP) {
             return "2020-05-26T07:59:58.000";
           }
+          // Intentional fall through
```
Comment (wcarlson5) [inline ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/CastEvaluatorTest.java:732]: Do we need this?

**Review by wcarlson5 — APPROVED** [review]
LGTM :+1

File: ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/CastEvaluatorTest.java (hunk 3 above)
Comment (mjsax) [reply ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/CastEvaluatorTest.java:732]: Not strictly. Saw it in another test with the same warning. -- Guess it's good to have for clarification.

**Review by guozhangwang — APPROVED** [review]

**Review by jzaralim — APPROVED** [review]

**Pull request merged**